mycity/platforms/amazon/lambda/custom/lambda_function.py

```diff
@@ -0,0 +1,106 @@
+"""
+Boston Data Alexa skill.
+
+This module is the entry point for processing voice data from an Alexa device.
+"""
+
+from mycity.mycity_data_model import MyCityDataModel
+from mycity.mycity_controller import MyCityController
+
+
+def lambda_handler(event, context):
+    """
+    Translate the Amazon request to a MC_Request_Model and call main.
+    """
+    print(
+        "[module: lambda_function]",
+        "[function: lambda_handler]",
+        "Amazon request received:\n",
+        str(event)
+    )
+
+    model = platform_to_mcd(event)
+    controller = MyCityController(model)
+    return mcd_to_platform(controller.main())
+
+
+def platform_to_mcd(event):
+    """
+    Translates from Amazon platform request to MyCityDataModel
+
+    :param event:
+    :return:
+    """
+    print(
+        "\n\n[module: lambda_function]",
+        "[function: platform_to_mcd]",
+        "Amazon request received:\n",
+        str(event)
+    )
+    mcd = MyCityDataModel()
+    mcd.request_type = event['request']['type']
+    mcd.request_id = event['request']['requestId']
+    mcd.is_new_session = event['session']['new']
+    mcd.session_id = event['session']['sessionId']
+    if 'attributes' in event['session']:
+        mcd.session_attributes = event['session']['attributes']
+    else:
+        mcd.session_attributes = {}
+    mcd.application_id = event['session']['application']['applicationId']
+    if 'intent' in event['request']:
+        mcd.intent_name = event['request']['intent']['name']
+        if 'slots' in event['request']['intent']:
+            mcd.intent_variables = event['request']['intent']['slots']
+    else:
+        mcd.intent_name = None
+    mcd.output_speech = None
+    mcd.reprompt_text = None
+    mcd.should_end_session = False
+    # if intent_variables was not set above, set it to an empty object here
+    mcd.intent_variables = mcd.intent_variables or {}

```

Comment:
Do we need to do this? It looks like `MyCityDataModel`'s constructor sets this to {}

Reply:
We do not. This is a relic from before I put that in the constructor.